internal/chain/transaction.go

```diff
@@ -137,21 +144,21 @@ func (b *TxBuild) buildLegacyTx(ctx context.Context, to *common.Address, value *
 	}), nil
 }
 
-func (b *TxBuild) getAndIncrementNonce() uint64 {
-	return atomic.AddUint64(&b.nonce, 1) - 1
-}
-
-func (b *TxBuild) refreshNonce(ctx context.Context) {
-	nonce, err := b.client.PendingNonceAt(ctx, b.Sender())
-	if err != nil {
-		log.WithFields(log.Fields{
-			"address": b.Sender(),
-			"error":   err,
-		}).Error("failed to refresh account nonce")
-		return
+func (b *TxBuild) getNextNonce(ctx context.Context) (uint64, error) {
+	b.nonceMu.Lock()
+	defer b.nonceMu.Unlock()
+	b.nonce++
+	// fetch from RPC every n txs, or after refresh interval - whichever is hit first
+	if time.Since(b.lastRefreshTime) > b.nonceRefreshInterval || b.nonce%b.nonceRefreshEvery == 0 {
```

Comment:
If nonces are fetched quite often (vs incrementing) and the load is high, it could be that `b.nonce%b.nonceRefreshEvery == 0` is unlikely to happen often (nonce 798 updates to 803 on fetch). Do we want to refetch nonce once in a while but a timeout is not sufficient?

Reply:
I think this is ok - after discussion on slack will close here :)